src/coreclr/jit/gentree.h

```diff
@@ -7651,16 +7651,20 @@ inline bool GenTree::IsSIMDZero() const
 //
 inline bool GenTree::IsFloatPositiveZero() const
 {
-    return IsCnsFltOrDbl() && !IsCnsNonZeroFltOrDbl();
+    if (IsCnsFltOrDbl())
```

Comment:
why is this change? Seems `IsCnsNonZeroFltOrDbl` already does this?

Reply:
It does, but we decided that `!IsCnsNonZeroFltOrDbl()` is a bit confusing; an explicit implementation seemed to be clearer.

Reply:
Perhaps add a comment explaining why you're not using `!IsCns ...`